reporting/mapreduce_logfile.py

```diff
@@ -0,0 +1,782 @@
+    try:
+        return substring_between(request, 'orders/', '-')
+    except ValueError:
+        return fail_to_parse('user_email', line)
+
+
+def get_scene_id(line):
+    try:
+        response_after_orderid = substring_between(line, 'orders/', '" ')
+        return substring_between(response_after_orderid, '/', '.tar.gz')
+    except ValueError:
+        try:
+            response_after_orderid = substring_between(line, 'orders/', '" ')
+            return substring_between(response_after_orderid, '/', '.cksum')
+        except ValueError:
+            return fail_to_parse('sceneid', line)
+
+
+def get_order_id(line):
+    request = substring_between(line, '] "', '" ')
+    try:
+        return substring_between(request, 'orders/', '/')
+    except ValueError:
+        return fail_to_parse('orderid', line)
+
+'''
+        Filter helper functions
+'''
+
+
+def n_filters(filters, iterable):
+        return (t for t in iterable if all(f(t) for f in filters))
+
+
+def is_successful_request(line):
+    return (get_rtcode(line) in ['200', '206'])
+
+
+def is_404_request(line):
+    return (get_rtcode(line) in ['404'])
+
+
+def is_aborted_request(line):
+    return (get_rtcode(line) in ['499', '304'])
+
+
+def is_production_order(line):
+    return ('"GET /orders/' in line)
+
+
+def is_dswe_order(line):
+    return (('"GET /provisional/dswe/' in line) or
+            ('"GET /downloads/provisional/dswe/' in line))
+
+
+def is_burned_area_order(line):
+    return (('"GET /provisional/burned_area/' in line) or
+            ('"GET /downloads/provisional/burned_area/' in line))
+
+
+def is_ordered_by_usgs_gov_email(line):
+    return ('@usgs.gov' in get_user_email(line))
+
+
+def is_ordered_through_ee_interface(line):
+    '''Use orderid to dtermine if ee interface was used to order
+
+    Precondition: Assumes earth explorer's orderid contains two dashes
+    Postcondition: Returns true if the ee explorer was used for the order.
+    '''
+    return (2 == get_order_id(line).count('-'))
+
+
+def is_within_daterange(line, start_date, end_date):
+    line_date = get_datetime(line)
+    return ((start_date < line_date) and (line_date < end_date))
+
+
+def is_in_this_month(line, month=datetime.datetime.now().month):
+    return (get_datetime(line).month == month)
+
+
+'''
+        Mappers
+'''
+
+
+def map_line_to_custom_tuple(line):
+    '''Extracts values from a line of text into tuple
+
+    Precondition: line is a ' ' separated list of data.
+        Preconditions for the following functions must also
+            be satisfied: get_user_email, get_bytes, get_order_id, get_scene_id
+    Postcondition: return tuple where len(tuple)==6
+    '''
+    remote_addr = line.split(' - ', 1)[0]
+    dt = get_datetime(line).isoformat()
+    user_email = get_user_email(line)
+    bytes_sent = get_bytes(line)
+    orderid = get_order_id(line)
+    sceneid = get_scene_id(line)
+    return (dt, remote_addr, user_email, orderid, sceneid, bytes_sent)
```

Comment:
I believe the python NamedTuple is practically declaring a class with defined functions. If it were used then it would it would require both the mapper and reducer to use the same NamedTuple in order to interpret the data.

Comments aside; I would be happy to make this change if you would like. I think NamedTuples make code more readable.